src/app/locales/vi_VN.ts

```diff
@@ -0,0 +1,513 @@
+  npm: 'npm',
+  yarn: 'yarn',
+  'Show me how to update!': 'Hướng dẫn tôi update!',
+  'Before Updating': 'Trước khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any changes needed before moving between these versions.':
+    'Hiện tại thì bạn không cần phải chuẩn bị gì trước khi update giữa những phiên bản này.',
+  'During the Update': 'Trong khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any recommendations for moving between these versions.':
+    'Hiện tại thì chúng tôi không có những gợi ý gì cho bạn cho việc update giữa những phiên bản này.',
+  'After the Update': 'Sau khi Update',
+  // tslint:disable-next-line: quotemark
+  'There aren\'t currently any changes needed after moving between these versions.':
+    'Hiện tại thì bạn không cần phải làm thêm gì sau khi update giữa những phiên bản này.',
+};
+
+const steps: LocalizedSteps = {
+  'Extends OnInit': {
+    action:
+      'Bảo đảm là bạn không dùng `extends OnInit`, hoặc dùng `extends` đối với bất kỳ LifeCycle nào. Thay vào đó, dùng `implements <LifeCycle>',
+  },
+  'Deep Imports': {
+    action:
+      'Dừng việc sử dụng Deep Imports, những biểu tượng này đều được đánh dấu với ký tự `ɵ` và không nằm trong API được public ra ngoài.',
+  },
+  invokeElementMethod: {
+    action:
+      'Dừng việc sử dụng hàm `Renderer.invokeElementMethod` vì hàm này đã bị xoá bỏ. Hiện tại không có sự thay thế nào.'
+  },
+  'Non Animations Module': {
+    action:
+      'Nếu bạn dùng Animations trong ứng dụng của bạn, bạn nên import `BrowserAnimationsModule` từ `@angular/platform-browser/animations` trong `NgModule` đầu vào của ứng dụng của bạn. (`AppModule`)'
+  },
+  'Native Form Validation': {
+    action:
+      'Angular đã bắt đầu thêm vào thuộc tính `novalidate` cho các nguyên tố liên quan đến Form khi bạn thêm `FormsModule` vào ứng dụng. Để sử dụng những hành vi nguyên bản của Forms (HTML), dùng `ngNoForm` hoặc thêm `ngNativeValidate`.'
+  },
+  RootRenderer: {
+    action: 'Thay thế `RootRenderer` bằng `RenderFactoryV2`.'
+  },
+  downgradeInjectable: {
+    action: 'Giá trị trả về của `upgrade/static/downgradeInjectable` đã thay đổi.',
+  },
+  'Animations Tests': {
+    action:
+      'Nếu bạn dùng Animations và có tests trong ứng dụng, thêm `mods[1].NoopAnimationsModule` vào `TestBed.initTestEnvironment` khi gọi hàm này.',
+  },
+  DefaultIterableDiffer: {
+    action: 'Dừng việc sử dụng `DefaultIterableDiffer`, `KeyValueDiffers#factories`, hoặc `IterableDiffers#factories`',
+  },
+  'Template Tag': {
+    action: 'Đổi tên các nhãn `template` thành `ng-template`',
+  },
+  OpaqueToken: {
+    action: 'Thay thế `OpaqueToken` bằng `InjectionToken`.',
+  },
+  DifferFactory: {
+    action: 'Nếu bạn có gọi hàm `DifferFactory.create(...)`, hãy bỏ tham số `ChangeDetectorRef` đi.',
+  },
+  'ErrorHandler Parameter': {
+    action: 'Dừng việc truyền tham số vào cho constructor của `ErrorHandler`',
```

Comment:
constructor may be confusing, NIT

Reply:
You have any suggestions for "constructor"?

Reply:
"hàm khởi tạo"?

Reply:
I think we don't need to translate all 100% to Vietnamese.  Some terms should be easier to understand in English than translated into Vietnamese. Example: `constructor`, `public API` ...
@nartc  How do you think?

Reply:
changed to use `hàm khởi tạo` for `constructor`. `public API` và `import` không translate.